### PR TITLE
Refactor regressions

### PR DESCRIPTION
**Pull request by ajreynol**
Moves regressions from regress0 to regress1 if debug version of cvc4 takes >.1 second.
Moves regressions from ( regress0 + regress1 ) to regress2 if debug version of cvc4 takes >2 seconds.

My scripts happened to reenable some regressions in the process.

Some regressions that were included in the distribution but disabled remained disabled.

### CODE REVIEW

**Comment by timothy-king** [comment]
The changes I am looking for have gone in. I am removing the block.

**Comment by ajreynol** [comment]
This is ready to review.

I changed the thresholds and did more moves regress0 -> regress1 and regress1 -> regress2.
I considered the time it takes a static debug binary of cvc4 to solve each regression with the proper options, without --check-model or --check-unsat-cores.
regress0 : solved <=.01 seconds
regress1 : solves <=1 second
  
A few other changes: I removed several deprecated benchmarks (mostly from the rewriterules directory) that weren't correct syntax.

One directory from regress0 was removed since all benchmarks moved to regress1.

make regress debug takes 1 min 30 on my laptop, 
make regress debug takes 45 seconds on the lab machines at Iowa,
make regress debug -j8 it takes 25 seconds on the lab machines at Iowa,
make regress production -j8 takes 14 seconds on the lab machines at Iowa

File: test/regress/regress0/rewriterules/Makefile.am [hunk 1]
```diff
@@ -23,14 +23,12 @@ MAKEFLAGS = -k
...
 	length_trick.smt2 \
-	length_gen_020.smt2 \
 	datatypes.smt2 \
-	datatypes_sat.smt2 \
 	relation.smt2 \
 	simulate_rewriting.smt2 \
 	native_arrays.smt2 
 
-# length_trick2.smt2 reachability_bbttf_eT_arrays.smt2 set_A_new_fast_tableau-base.smt2 set_A_new_fast_tableau-base_sat.smt2
+# length_trick2.smt2 reachability_bbttf_eT_arrays.smt2 REM REM
```
Comment (aniemetz) [inline test/regress/regress0/rewriterules/Makefile.am]: Are these REM markers intentional?

Comment (ajreynol) [reply test/regress/regress0/rewriterules/Makefile.am]: No, I'll remove.

File: test/regress/regress0/sygus/Makefile.am [hunk 2]
```diff
@@ -18,79 +18,28 @@ MAKEFLAGS = -k
...
-
 # sygus tests currently taking too long for make regress
 EXTRA_DIST = $(TESTS) \
-  max.smt2 \
-  sygus-uf.sl \
-  enum-test.sy 
+  sygus-uf.sl
   
-# strings-concat-3-args.sy
+# REM
```
Comment (aniemetz) [inline test/regress/regress0/sygus/Makefile.am]: Found another one.

**Pull request merged**